test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -449,12 +459,103 @@ defmodule Trento.SapSystemTest do
           sap_system_id: sap_system_id,
           sid: sid
         ),
-        %{instance_number: instance_number, host_id: host_id} =
-          build(:application_instance_registered_event,
+        build(:application_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          features: "MESSAGESERVER"
+        ),
+        build(:application_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          features: "ABAP",
+          instance_number: "10",
```

Comment:
You should use the `instance_number` variable here I guess. 
The same for the `RegisterApplicationInstance.new` function below